start-kafka.sh

```diff
@@ -3,7 +3,9 @@
 if [[ -z "$KAFKA_PORT" ]]; then
     export KAFKA_PORT=9092
 fi
-if [[ -z "$KAFKA_ADVERTISED_PORT" ]]; then
+if [[ -z "$KAFKA_ADVERTISED_PORT" && \
+  -n "$KAFKA_LISTENERS" && \
+  -n "$KAFKA_ADVERTISED_LISTENERS" ]]; then
```

Comment:
Just realized, I think I intended these to be `-z`: don't default the `_PORT` unless none of these values are present. Will update.